lib/class-wp-widget-block.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Widget API: WP_Widget_Block class
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Core class used to implement a Block widget.
+ *
+ * @see WP_Widget
+ */
+class WP_Widget_Block extends WP_Widget {
+
+	/**
+	 * Default instance.
+	 *
+	 * @since 4.8.1
+	 * @var array
+	 */
+	protected $default_instance = array(
+		'content' => '',
+	);
+
+	/**
+	 * Sets up a new Block widget instance.
+	 *
+	 * @since 4.8.1
+	 */
+	public function __construct() {
+		$widget_ops  = array(
+			'classname'                   => 'widget_block',
+			'description'                 => __( 'Gutenberg block.', 'gutenberg' ),
+			'customize_selective_refresh' => true,
+		);
+		$control_ops = array(
+			'width'  => 400,
+			'height' => 350,
+		);
+		parent::__construct( 'block', __( 'Gutenberg Block', 'gutenberg' ), $widget_ops, $control_ops );
+	}
+
+	/**
+	 * Outputs the content for the current Block widget instance.
+	 *
+	 * @since 4.8.1
+	 *
+	 * @global WP_Post $post Global post object.
+	 *
+	 * @param array $args     Display arguments including 'before_title', 'after_title',
+	 *                        'before_widget', and 'after_widget'.
+	 * @param array $instance Settings for the current Block widget instance.
+	 */
+	public function widget( $args, $instance ) {
+		echo do_blocks( $instance['content'] );
```

Comment:
This will sometimes render a `<form>` (e.g. Search block), which is invalid HTML, because we'll then have a `<form>` nested within a `<form>`.

Reply:
@noisysocks I don't see a `<form>` anywhere in this file:

https://github.com/WordPress/gutenberg/tree/master/packages/block-library/src/search/edit.js

Where I can see a `<form>` is in the `legacy-widget` block:

https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/legacy-widget/edit/dom-manager.js#L63

I don't think that's a problem though - the widget screen is not wrapped in a `<form>` and so each block is free to render anything it wants.

Reply:
Sorry, I wasn't clear here. I'm referring to the legacy `widgets.php` screen. By default every widget is wrapped with a `<form>`.

https://github.com/WordPress/wordpress-develop/blob/master/src/wp-admin/includes/widgets.php#L205

This means that you can sometimes end up with invalid HTML.

1. Navigate to _Widgets (beta)_ and add a Search block to one of the block areas.
2. Navigatie to _Widgets_ (the old screen).
3. View the page source and search for `wp-block-search`.

You'll see that there's a `<form>` inside a `<form>`. Firefox helpfully highlights the invalid HTML.

<img width="1440" alt="Screen Shot 2020-08-26 at 11 05 17" src="https://user-images.githubusercontent.com/612155/91242962-190de480-e78c-11ea-9214-4f7b9dff7edc.png">